ImageInLib/ImageInLib/src/data_load.c

```diff
@@ -133,18 +134,25 @@ bool load3dDataArrayRAW(dataType ** imageDataPtr, const size_t imageLength, cons
 
 					if (dType == BINARY_DATA)
 					{
-						value = getc(file);
+						fread(&value, sizeof(dataType), 1, file);
 						imageDataPtr[k][xd] = (dataType)value;
 					}
 					else if (dType == ASCII_DATA)
 					{
-						fscanf_s(file, "%d", &value);
+						fscanf_s(file, "%f", &value);
 						imageDataPtr[k][xd] = (dataType)value;
 					}
 				}
 			}
 		}
 	}
+
+	//switch the two first bytes
+	for (k = 0; k < imageHeight; k++) {
+		for (i = 0; i < imageLength * imageWidth; i++) {
+			reverBytes(&imageDataPtr[k][i], sizeof(dataType));
+		}
+	}
 	fclose(file);
```

Comment:
This always being done to any loaded data, what if it does not need revertBytes? Add a check to determine if the data needs to revertBytes (Big endian or Little endian).